lib/si-split-graph/src/tests/mod.rs

```diff
@@ -303,20 +267,486 @@ fn ordered_container() -> SplitGraphResult<()> {
-    let reader_writer = TestReadWriter {
-        graphs: HashMap::new(),
+fn default_edges() -> SplitGraphResult<()> {
+    let mut split_graph: SplitGraph<TestNodeWeight, TestEdgeWeight, TestEdgeWeightDiscriminants> =
+        SplitGraph::new(3);
+
+    let nodes: Vec<_> = ["a", "b", "c", "d", "e", "f"]
+        .into_iter()
+        .map(|name| TestNodeWeight {
+            id: Ulid::new(),
+            lineage_id: SplitGraphNodeId::new(),
+            name: name.to_string(),
+            merkle_tree_hash: MerkleTreeHash::nil(),
+        })
+        .collect();
+
+    let mut node_id_map = HashMap::new();
+    for node in &nodes {
+        split_graph.add_or_replace_node(node.clone())?;
+        node_id_map.insert(node.name.as_str(), node.id);
+        if node.name == "a" {
+            split_graph.add_edge(split_graph.root_id()?, TestEdgeWeight::EdgeA, node.id)?;
+        } else {
+            split_graph.add_edge(
+                node_id_map.get(&"a").copied().unwrap(),
+                TestEdgeWeight::EdgeB { is_default: false },
+                node.id,
+            )?;
+        }
+    }
+
+    let a_id = node_id_map.get(&"a").copied().unwrap();
+    let b_id = node_id_map.get(&"b").copied().unwrap();
+
+    split_graph.cleanup_and_merkle_tree_hash();
+    let mut updated_graph = split_graph.clone();
+    updated_graph.remove_edge(a_id, TestEdgeWeightDiscriminants::EdgeB, b_id)?;
+    updated_graph.add_edge(a_id, TestEdgeWeight::EdgeB { is_default: true }, b_id)?;
+    updated_graph.cleanup_and_merkle_tree_hash();
+
+    fn default_edge_assertions(
+        graph: &SplitGraph<TestNodeWeight, TestEdgeWeight, TestEdgeWeightDiscriminants>,
+        from_id: SplitGraphNodeId,
+        default_target: SplitGraphNodeId,
+    ) -> SplitGraphResult<()> {
+        let edges: Vec<_> = graph
+            .edges_directed(from_id, Outgoing)?
+            .map(|edge| (edge.weight().clone(), edge.target()))
+            .collect();
+
+        assert_eq!(5, edges.len());
+        let mut hit_default = false;
+        for edge in edges {
+            if edge.1 != default_target {
+                assert!(matches!(
+                    edge.0,
+                    TestEdgeWeight::EdgeB { is_default: false }
+                ));
+            } else {
+                assert!(
+                    matches!(edge.0, TestEdgeWeight::EdgeB { is_default: true }),
+                    "{:?} should be default target",
+                    default_target
+                );
+                hit_default = true;
+            }
+        }
+
+        assert!(hit_default);
+        Ok(())
+    }
+
+    default_edge_assertions(&updated_graph, a_id, b_id)?;
+
+    let updates = split_graph.detect_updates(&updated_graph);
+    split_graph.perform_updates(&updates);
+    split_graph.cleanup_and_merkle_tree_hash();
+    default_edge_assertions(&split_graph, a_id, b_id)?;
+
+    Ok(())
+}
+
+#[test]
+fn cross_graph_node_id_updates() -> SplitGraphResult<()> {
```

Comment:
Very great work. I was curious about how this would work and I can see it clearly. Really great tests